src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -2384,6 +2384,11 @@ public void Paste()
         /// directly in this point. </param>
         public void Paste(Point2D targetPoint, bool useOffset = true)
         {
+            //When called from somewhere other than StateMachine and only ConnectorPins are selected.
+            if (!ClipBoard.Where(m => !(m is ConnectorPinModel)).Select(m => m).Any())
```

Comment:
Maybe use `All()` instead of the more complex way you are using?

Reply:
Here that makes sense. Fixed.